docs/proposal-partially-satisfied-prerequisites.md

```diff
@@ -0,0 +1,347 @@
+
+We should also improve the visibility of stalled workflows and blocked tasks.
+*  `cylc scan` and the UI gscan panel should clearly indicate if a running workflow is stalled or has blocked tasks.
+* Similarly for stopped workflows (also other states: "not started", "run to completion", "died"?, etc).
+* We need a mechanism to ensure that any workflows which have stopped unexpectedly don't get missed and that users can easily find their most recent workflows
+  (e.g. cylc review orders by last activity).
+
+## Unhandled task failures
+
+Does the new syntax mean we can simplify our handling of unhandled task failures?
+
+In the simple case of `foo => bar`, if `foo` fails we'll be able to identify the unsatisfied `bar` as a problem even if we don't keep the failed `foo` in the pool.
+However, keeping it in the pool (n=0) is likely to be more convenient and consistent.
+In any case, what happens if `bar` fails - we certainly need to keep that in the pool.
+
+So, we need to continue to keep any failed tasks in the pool (and consider them "blocked") unless they have a failure trigger.
+
+**Question**:
+How do we cater for cases where task failure is accepted and does not require any action?
+We will need to contine to support suicide triggers for Cylc 7 compatibility so we can simply use:
+
+```
+        foo:fail => !foo
+```
+
+Alternatively we could introduce a special syntax for this such as:
+
+```
+        foo:fail => NULL
+```
+
+Note that a special syntax makes it easier to deal with this case:
+
+```
+        a | b => c
+        a:fail | b:fail => NULL
+```
+
+In this case we need `a` or ` b` to succeed but we don't mind if one of them fails.
+If both fail we don't get the convenience of them remaining in the task pool but at least `c` will be blocked so `a` and `b` will be visible in n=1.
+
+## How do we tell if it is not possible for the task pre-requisities to be met?
+
+Once all of the tasks referenced in the pre-requisities have completed then clearly we know whether it is possible for the pre-requisities to be met.
+However, what happens if some of those tasks never run.
+Consider this example again:
+
+```
+        foo ?=> bar => qux
+        foo:fail ?=> baz => qux
+```
+
+If, for example, `foo` succeeds then `qux` becomes stuck waiting for `baz` to run.
+In theory we can tell that `baz` cannot run but this would require graph traversal which could be complicated.
+For the moment we will not attempt this.
+
+One problem with this is that, even with required tasks, the workflow may never stall (depending on how the runahead limit is set).
+However the `waiting` tasks should at least be visible to the user and there should also be another visible cause (blocked or failed tasks) unless there is an error in the graph.
+
+## Cylc 7 compatibility
+
+Cylc 7 workflows will not be taking advantage of the new syntax which means there will only be required triggers.
+However there should always be corresponding suicide triggers where needed.
```

Comment:
Note this section gets considerably easier in #129 because alternate branches don't get spawned at all (even as unsatisfied prerequisites) so they don't need to be suicided.